src/machines/java/com/enderio/machines/client/gui/screen/OmniBufferScreen.java

```diff
@@ -0,0 +1,86 @@
+    private EnergyTextboxWidget output;
+
+    public OmniBufferScreen(OmniBufferMenu pMenu, Inventory pPlayerInventory, Component pTitle) {
+        super(pMenu, pPlayerInventory, pTitle);
+    }
+
+    @Override
+    protected void init() {
+        super.init();
+
+        addRenderableOnly(addRenderableOnly(new CapacitorEnergyWidget(this, getMenu().getBlockEntity()::getEnergyStorage, getMenu().getBlockEntity()::isCapacitorInstalled, 16 + leftPos, 14 + topPos, 9, 42)));
+
+        addRenderableWidget(new EnumIconWidget<>(this, leftPos + imageWidth - 8 - 12, topPos + 6, () -> menu.getBlockEntity().getRedstoneControl(),
+            control -> menu.getBlockEntity().setRedstoneControl(control), EIOLang.REDSTONE_MODE));
+
+        addRenderableWidget(new IOConfigButton<>(this, leftPos + imageWidth - 6 - 16, topPos + 22, 16, 16, menu, this::addRenderableWidget, font));
+
+        input = new EnergyTextboxWidget(this, () -> this.getMenu().getBlockEntity().getEnergyStorage().getMaxEnergyUse(), this.font, leftPos + 33, topPos + 18, 49, this.font.lineHeight + 2, Component.empty());
+        input.setValue(input.formatEnergy(Integer.toString(this.getMenu().getBlockEntity().getMaxInput())));
+        addRenderableWidget(input);
+
+        output = new EnergyTextboxWidget(this, () -> this.getMenu().getBlockEntity().getEnergyStorage().getMaxEnergyUse(), this.font, leftPos + 33, topPos + 48, 49, this.font.lineHeight + 2, Component.empty());
+        output.setValue(output.formatEnergy(Integer.toString(this.getMenu().getBlockEntity().getMaxOutput())));
+        addRenderableWidget(output);
+
+    }
+
+    @Override
+    public ResourceLocation getBackgroundImage() {
+        return BG_TEXTURE;
+    }
+
+    @Override
+    protected Vector2i getBackgroundImageSize() {
+        return new Vector2i(175, 166);
+    }
+
+    @Override
+    public void render(GuiGraphics guiGraphics, int pMouseX, int pMouseY, float pPartialTicks) {
+        super.render(guiGraphics, pMouseX, pMouseY, pPartialTicks);
+        PowerBufferBlockEntity be = this.getMenu().getBlockEntity();
+        guiGraphics.drawString(font, "Input:", leftPos + 33, topPos + 18 - font.lineHeight - 2, 1, false);
+        guiGraphics.drawString(font, "Output:", leftPos + 33, topPos + 48 - font.lineHeight - 2, 1, false);
+    }
+
+
+    //Reset values if no capacitor, else display the value stored in the power buffer
+    @Override
+    protected void containerTick() {
+        super.containerTick();
+        PowerBufferBlockEntity be = this.getMenu().getBlockEntity();
+        if (be != null) {
+            if (be.requiresCapacitor() && be.isCapacitorInstalled()) {
+                be.setMaxInput(input.getInteger());
+                be.setMaxOutput(output.getInteger());
+            } else {
+                be.setMaxInput(0);
+                input.setValue("0");
+
+                be.setMaxOutput(0);
+                output.setValue("0");
+            }
+        }
+    }
```

Comment:
You can probably use `onInventoryContentsChanged` on BE instead of containerTick

Reply:
I don't think so, we are not updating the inventory when updating I/O rates

Reply:
my bad. use Editbox setResponder()

Reply:
I've implemented a version with the setResponder().
However now it updates IO every tick but I want them to be updated if and only if the textbox is not selected.